lib/simplifyExpression/distributeSearch/index.js

```diff
@@ -7,8 +7,7 @@ const ChangeTypes = require('../../ChangeTypes');
 const Negative = require('../../Negative');
 const Node = require('../../node');
 const TreeSearch = require('../../TreeSearch');
-
-const search = TreeSearch.postOrder(distribute);
+const search = TreeSearch.preOrder(distribute);
```

Comment:
The changes that I make to `distributeSearch` in this pull request requires us to traverse the tree in a postOrder fashion starting from the root node down to the leafs. This way we can catch the higher order operations that should be evaluated first. For example in `(x (x + 1))^2`, we want to distribute the exponent first before distributing the inner multiplication. Let me know if this is still confusing!